test/fuzz/classifier-spdx.c

```diff
@@ -0,0 +1,76 @@
+#include "../licences.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+const size_t CHUNK_SIZE = 4096;
+
+struct buffer {
+	char* data;
+	size_t capacity;
+	size_t size;
+};
+
+void buffer_cleanup(struct buffer* buf) {
+	free(buf->data);
+	memset(buf, 0, sizeof(*buf));
+}
+
+int buffer_append(struct buffer* buf, const char* data, size_t size) {
+	if (buf->size + size > buf->capacity) {
+		size_t new_capacity = buf->size + size;
+		if (new_capacity % CHUNK_SIZE) {
+			new_capacity = ((new_capacity / CHUNK_SIZE) + 1) * CHUNK_SIZE;
+		}
+		char* new_data = realloc(buf->data, new_capacity);
+		if (!new_data) {
+			return -1;
+		}
+		buf->data = new_data;
+		buf->capacity = new_capacity;
+	}
+
+	memcpy(&buf->data[buf->size], data, size);
+	buf->size += size;
+	return 0;
+}
+
+char* read_stdin() {
+	struct buffer buf = {0};
+
+	while (!feof(stdin)) {
+		char chunk[CHUNK_SIZE];
+		size_t read = fread(chunk, 1, sizeof(chunk), stdin);
+		if (buffer_append(&buf, chunk, read)) {
+			buffer_cleanup(&buf);
+			return NULL;
+		}
+	}
+	buffer_append(&buf, "", 1);
+	return buf.data;
+}
+
+int main(int argc, char* argv[]) {
```

Comment:
I feel this could be more useful if it was modified slightly to test only one classifier at a time, and take it as an argument - i.e. take a look at `argv[1]` and based on that use the proper `->*Classifier`. We could then use Make targets for convenience, e.g. introduce a `fuzz-classifier-spdx-strict` target that'd call this program with the proper arguments.

Reply:
On the other hand, it's easier to evaluate coverage completeness of the fuzzing process - the lenient/strict distinction is just a handful of if statements.

Anyway, done, with loose classifier support, because why not.